### PR TITLE
Add initial doc on user roles

### LINKED ISSUE #27: Create doc on user roles
Describe the user roles of the TRE

- TRE Developer
- TRE Service Integrator
- Azure Administrator
- TRE Administrator
- TRE Workspace owner
- Researcher

Related to #16

### PR DESCRIPTION
**Pull request by christoferlof**
Closes #27 - Initial doc on the user roles.

### CODE REVIEW

File: docs/user-roles.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+## TRE administrator
+
+Day-to-day running and operations of the Azure TRE instance without touching Azure resources.
+
+Example tasks:
+
+- Manage workspace owner users.
+- Provision workspaces
+- Manage shared services. (e.g. Avaiable packages in Package mirror shared service.)
```
Comment (TessFerrandez) [inline docs/user-roles.md:51]:
```suggestion
- Manage shared services. (e.g. available packages in package mirror shared service.)
```

File: docs/user-roles.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+Before you deploy a Trusted Research Environment based on the Azure TRE solution, you should consider your scenario and have an understanding of which of these roles that needs to be staffed.
+
+## Role overview
+
+While we have defined 8 different user roles for the Azure TRE solution, not all of them are required in all scenarios. Three of the roles are currently available for Role Bases Access Control within the TRE.  
+
+| Role | Key task | TRE RBAC |
+|------|----------|----------|
+| Azure administrator | Deploy the TRE | |
+| TRE administrator | Administrate the TRE | ✔ |
```
Comment (TessFerrandez) [inline docs/user-roles.md:14]:
```suggestion
| TRE administrator | Administer the TRE | ✔ |
```

File: docs/user-roles.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+| TRE workspace owner | Own a workspace | ✔ |
```
Comment (TessFerrandez) [inline docs/user-roles.md:15]: Is the key task owning the workspace or being the point of contact?

File: docs/user-roles.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+| Researcher | Perform research on the data | ✔ |
+| TRE service integrator | Integrate additional workspace services | |
+| Azure TRE developer | Extend the TRE OSS solution | |
+| Data engineer | Move data to the TRE | |
```
Comment (TessFerrandez) [inline docs/user-roles.md:19]: and from? below it mentions that the owner also imports and exports data?

File: docs/user-roles.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+- Manage workspace owner users.
+- Provision workspaces
+- Manage shared services. (e.g. Avaiable packages in Package mirror shared service.)
+- Monitor workspace usage and billing.
+- Set and manage quotas.
+
+Expected skills:
+
+- Limited or no Azure knowledge expected.
+- Python.
```
Comment (TessFerrandez) [inline docs/user-roles.md:58]: Why is python an expected skill here?

File: docs/user-roles.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+- Manage Researcher users.
+- Export data from workspace.
+- Import data and make it available within the workspace.
+- Enable services within the workspace.
+- Monitor billing and usage of the workspace.
+
+Expected skills:
+
+- Limited or no Azure knowledge expected.
+- Python.
```
Comment (TessFerrandez) [inline docs/user-roles.md:75]: Same question as above

File: docs/user-roles.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+- Git
+- Linux
+
+## TRE service integrator
+
+Integrates workspace service types with an Azure TRE instance. This involves extending the Azure Infrastructure as Code templates to make a workspace service available within an Azure TRE instance.
+
+Example tasks:
+
+- Integrate a workspace service type with your Azure TRE instance
```
Comment (TessFerrandez) [inline docs/user-roles.md:98]: maybe also Implement Infrastructure as Code templates for new workspace servcies

File: docs/user-roles.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+Software developer who contributes to the development of the Azure TRE solution.
+
+Example tasks:
+
+- Modify the deployment service, management API and other components of the AzTRE solution.
+- Contribute to the Azure TRE OSS solution.
+
+Expected skills:
+
+- Git, GitHub
```
Comment (TessFerrandez) [inline docs/user-roles.md:117]: Seems like this person should also have the same skills as the TRE service integrator - so PowerShell, Bash, Azure Administration - etc.
For consistency maybe also list the skills in the same order so it is easy to find the differences

File: docs/user-roles.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+- Python, Bash, Linux
+- Azure Data Factory, Other ETL tools.
+
+## Information Security Officer
+
+Needs to understand the security posture of the TRE to ensure that the organization is compliant with the information governance framework and additional relevant regulations.
+
+Example tasks:
+
+- Use the AzTRE documentation to understand the security posture of the TRE.
```
Comment (TessFerrandez) [inline docs/user-roles.md:141]:
```suggestion
- Use the Azure TRE documentation to understand the security posture of the TRE.
```

**Review by TessFerrandez — DISMISSED** [review]

File: docs/user-roles.md (hunk 3 above)
Comment (christoferlof) [reply docs/user-roles.md:15]: I'd say owning and overall accountability of what's happening within the TRE workspace.

**Review by Lybecker — APPROVED** [review]

File: docs/user-roles.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
+# User roles
+
+The Azure TRE solution has 8 different user roles defined. The roles are modeled around a set of tasks for each role. The roles are not mutually exclusive, and one person can have multiple roles assigned to be able to carry out a broader set of tasks.
+
+Before you deploy a Trusted Research Environment based on the Azure TRE solution, you should consider your scenario and have an understanding of which of these roles that needs to be staffed.
+
+## Role overview
+
+While we have defined 8 different user roles for the Azure TRE solution, not all of them are required in all scenarios. Three of the roles are currently available for Role Bases Access Control within the TRE.
```
Comment (tompaana) [inline docs/user-roles.md:9]:
```suggestion
While we have defined 8 different user roles for the Azure TRE solution, not all of them are required in all scenarios. Three of the roles support role-based access control (RBAC) within the TRE.  
```

File: docs/user-roles.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+- PowerShell, Bash
+
+## TRE administrator
+
+Day-to-day running and operations of the Azure TRE instance without touching Azure resources.
+
+Example tasks:
+
+- Manage workspace owner users.
+- Provision workspaces
```
Comment (tompaana) [inline docs/user-roles.md:50]:
```suggestion
- Provision workspaces.
```

File: docs/user-roles.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+- Manage shared services. (e.g. avaiable packages in package mirror shared service.)
```
Comment (tompaana) [inline docs/user-roles.md:51]:
```suggestion
- Manage shared services e.g., available packages in package mirror shared service.
```

**Review by tompaana — CHANGES_REQUESTED** [review]
This looks like a good start! Check your punctuation.

**Review by tompaana — APPROVED** [review]

**Pull request merged**